ztp/ztp-ran-policy-generator/kustomize/plugin/ranPolicyGenerator/v1/ranpolicygenerator/policyGen/policyBuilder.go

```diff
@@ -0,0 +1,187 @@
+package policyGen
+
+import (
+	"io/ioutil"
+	utils "github.com/serngawy/cnf-features-deploy/ztp/ztp-ran-policy-generator/kustomize/plugin/ranPolicyGenerator/v1/ranpolicygenerator/utils"
+	//"fmt"
+	yaml "gopkg.in/yaml.v3"
+	"strings"
+	"reflect"
+)
+
+type PolicyBuilder struct {
+	RanGenTemp utils.RanGenTemplate
+	SourcePoliciesDir string
+}
+
+func NewPolicyBuilder(RanGenTemp utils.RanGenTemplate, SourcePoliciesDir string) *PolicyBuilder {
+	return &PolicyBuilder{RanGenTemp:RanGenTemp, SourcePoliciesDir:SourcePoliciesDir}
+}
+
+func (pbuilder *PolicyBuilder) Build() (map[string]interface{}) {
+
+	policies := make(map[string]interface{})
+	if len(pbuilder.RanGenTemp.SourceFiles) != 0 {
+		namespace, path, matchKey, matchValue, matchOper := pbuilder.getPolicyNsPath()
+		subjects := make([]utils.Subject , 0)
+		for id, sFile := range pbuilder.RanGenTemp.SourceFiles {
+			pname, rname := pbuilder.getPolicyName(id)
+			sPolicyFile, err := ioutil.ReadFile(pbuilder.SourcePoliciesDir + "/" + sFile.FileName + utils.FileExt)
+			if err != nil {
+				panic(err)
+			}
+			rname, resourceDef := pbuilder.getResourceDefinition(sFile.Spec, sPolicyFile, rname, pbuilder.RanGenTemp.Metadata.Labels.Mcp)
+
+			acmPolicy := pbuilder.getPolicy(pname + "-" + rname, namespace, resourceDef)
+			policies[path + "/" + pname + "-" + rname] = acmPolicy
+			subject := CreatePolicySubject(pname + "-" + rname )
+			subjects = append(subjects, subject)
+		}
+
+		placementRule := CreatePlacementRule(pbuilder.RanGenTemp.Metadata.Name, namespace, matchKey, matchOper, matchValue)
+		policies[path + "/" + placementRule.Metadata.Name] = placementRule
+
+		placementBinding := CreatePlacementBinding(pbuilder.RanGenTemp.Metadata.Name, namespace, placementRule.Metadata.Name, subjects)
+		policies[path + "/" + placementBinding.Metadata.Name] = placementBinding
+	}
+	return policies
+}
+
+func (pbuilder *PolicyBuilder) getPolicy(name string, namespace string, objMap map[string]interface{}) utils.AcmPolicy {
+	objTemp := CreateObjTemplates(objMap)
+	acmConfigPolicy := CreateAcmConfigPolicy(name)
+
+	objTempArr := make([]utils.ObjectTemplates, 1)
+	objTempArr[0] = objTemp
+	acmConfigPolicy.Spec.ObjectTemplates = objTempArr
+
+	policyObjDef := utils.PolicyObjectDefinition{}
+	policyObjDef.ObjDef = acmConfigPolicy
+
+	acmPolicy := CreateAcmPolicy(name, namespace)
+	policyObjDefArr := make([]utils.PolicyObjectDefinition, 1)
+	policyObjDefArr[0] = policyObjDef
+
+	acmPolicy.Spec.PolicyTemplates = policyObjDefArr
+	return acmPolicy
+}
+
+func (pbuilder *PolicyBuilder) getResourceDefinition(spec map[string]interface{}, sourcePolicy []byte, name string, mcp string) (string, map[string]interface{}) {
```

Comment:
A brief description on non-trivial methods which tells what the method consumes/produces is really helpful for maintaining the software (and for others to contribute).

Reply:
okay, actually the name should be CustomResource not ResourceDefinition